File: backend/app/models/reservation.py
from typing import Any, Dict, List, Optional
from datetime import datetime
from pydantic import Field
from app.models.base import BaseDocument
from uuid import uuid4

class Reservation(BaseDocument):
    id: str = Field(default_factory=lambda: uuid4().hex, alias="_id")
    user_id: str
    community_id: Optional[str] = None
    category: str
    title: str
    details: Dict[str, Any]
    assigned_to: List[str] = Field(default_factory=list)
    is_shared: bool = False
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    status: str

    class Settings:
        name = "reservations"

File: frontend/lib/domain/entities/reservation/reservation.dart
class Reservation {
  final String id;
  final String? userId;
  final String? communityId;
  final String? communityName;
  final List<String> assignedTo;
  final String category;
  final String title;
  final Map<String, dynamic> details;
  final bool isShared;
  final DateTime? startDate;
  final DateTime? endDate;
  final String status;

  Reservation({
    this.id = '',
    this.userId,
    this.communityId,
    this.communityName,
    this.assignedTo = const [],
    required this.category,
    required this.title,
    required this.details,
    this.isShared = false,
    this.startDate,
    this.endDate,
    required this.status,
  });
}
